docs/classes.rst

```diff
@@ -295,8 +295,13 @@ C++11-style strongly typed enumerations.
 
    When the annotation :cpp:class:`nb::is_arithmetic() <is_arithmetic>` is
    passed to :cpp:class:`nb::enum_\<T\> <enum_>`, the resulting Python type
-   will support arithmetic and bit-level operations like comparisons, and, or,
-   xor, negation, etc.
+   will support arithmetic and bit-level operations (and, or,
+   xor, negation).  The operands of these operations may be either enumerators
+   When the annotation :cpp:class:`nb::flag_enum() <flag_enum>` is passed to
+   :cpp:class:`nb::enum_\<T\> <enum_>`, the resulting Python type will be an
+   `enum.IntFlag`, meaning its enumerators can be combined using bitwise operators
+   in a type-safe way: the result will have the same enumeration type as the operands,
+   and only enumerators of the same type can be combined.
```

Comment:
I don't think this text is up-to-date wrt. the code. We can have combinations of `flag_enum` and `is_arithmetic`, which influences the base type and valid set of arithmetic operators for a total of 4 possibilities.